hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockDataStreamOutput.java

```diff
@@ -392,7 +398,8 @@ private void executePutBlock(boolean close,
     final List<StreamBuffer> byteBufferList;
     if (!force) {
       Preconditions.checkNotNull(bufferList);
-      byteBufferList = bufferList;
+      byteBufferList = buffersForPutBlock;
+      buffersForPutBlock = null;
       Preconditions.checkNotNull(byteBufferList);
```

Comment:
@sadanand48 , there is a NPE
```
java.lang.NullPointerException
at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:880)
at org.apache.hadoop.hdds.scm.storage.BlockDataStreamOutput.executePutBlock(BlockDataStreamOutput.java:403)
at org.apache.hadoop.hdds.scm.storage.BlockDataStreamOutput.handleFlush(BlockDataStreamOutput.java:489)
at org.apache.hadoop.hdds.scm.storage.BlockDataStreamOutput.close(BlockDataStreamOutput.java:509)
at org.apache.hadoop.ozone.client.io.BlockDataStreamOutputEntry.close(BlockDataStreamOutputEntry.java:122)
at org.apache.hadoop.ozone.client.io.KeyDataStreamOutput.handleStreamAction(KeyDataStreamOutput.java:460)
...
```
It seems the preconditions is incorrect.
```
      Preconditions.checkNotNull(byteBufferList);
```
Please take a look.  Thanks.

Reply:
Created PR #2984 to address this